examples/prefetch.rs

```diff
@@ -0,0 +1,50 @@
+extern crate shred;
+
+use shred::{DispatcherBuilder, Fetch, FetchMut, Prefetch, Resources, System, SystemData};
+
+#[derive(Debug)]
+struct ResA;
+
+#[derive(Debug)]
+struct ResB;
+
+struct PrintSystem<'a> {
+    prefetch: <(Fetch<'a, ResA>, FetchMut<'a, ResB>) as SystemData<'a>>::Prefetch,
+}
+
+impl<'a> PrintSystem<'a> {
+    fn new(res: &'a Resources) -> Self {
+        PrintSystem {
+            prefetch: Prefetch::prefetch(res),
+        }
+    }
+}
+
+impl<'a, 'b> System<'a> for PrintSystem<'b> {
+    type SystemData = (Fetch<'a, ResA>, FetchMut<'a, ResB>);
+
+    fn fetch_data(&mut self, res: &'a Resources) -> Self::SystemData {
+        self.prefetch.fetch()
+    }
+
+    fn run(&mut self, data: Self::SystemData) {
+        let (a, mut b) = data;
+
+        println!("{:?}", &*a);
+        println!("{:?}", &*b);
+
+        *b = ResB; // We can mutate ResB here
+        // because it's `FetchMut`.
+    }
+}
+
+fn main() {
+    let mut resources = Resources::new();
+    let mut dispatcher = DispatcherBuilder::new()
+        .add(PrintSystem::new(&resources), "print", &[]) // Adds a system "print" without dependencies
```

Comment:
That's the idea here: we get the cell from the resources on initialization, so there is no need for hash map access anymore.